### PR TITLE
Extract playlists description

### PR DESCRIPTION
**Pull request by ChunkyProgrammer**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [ ] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [ ] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

This PR adds a getDescription method for playlist extraction + tests for the description. I am not planning on creating a PR in NewPipe to support this (I implemented this to hopefully add this to the Piped API)

### API changes
This PR adds
- `PlaylistExtractor.getDescription()` and the corresponding  `PlaylistInfo.getDescription()`
- `PlaylistInfoItemExtractor.getDescription()` and the corresponding  `PlaylistInfoItem.getDescription()`

### CODE REVIEW

File: extractor/src/main/java/org/schabi/newpipe/extractor/playlist/PlaylistExtractor.java [hunk 1]
```diff
@@ -21,6 +22,8 @@ public PlaylistExtractor(final StreamingService service, final ListLinkHandler l
 
     public abstract long getStreamCount() throws ParsingException;
 
+    public abstract Description getDescription() throws ParsingException;
```
Comment (AudricV) [inline extractor/src/main/java/org/schabi/newpipe/extractor/playlist/PlaylistExtractor.java:25]: I think you should provide a default implementation returning `Description.EMPTY_DESCRIPTION`, like done for other methods below, as most of the implementations return this value.

**Review by AudricV — CHANGES_REQUESTED** [review]
Thanks, you need to also implement these changes in `PlaylistInfo`.

**Comment by TobiGr** [comment]
> Thanks, you need to also implement these changes in PlaylistInfo.

I already implemented that locally. There are a few more changes which I am going to push within the next days.

File: extractor/src/main/java/org/schabi/newpipe/extractor/playlist/PlaylistExtractor.java (hunk 1 above)
Comment (ChunkyProgrammer) [reply extractor/src/main/java/org/schabi/newpipe/extractor/playlist/PlaylistExtractor.java:26]: Would you still want this to be done? TobiGr added support for PeerTube and Soundcloud

**Comment by TobiGr** [comment]
@fynngodau I am unsure which parts of the album description should be used as description. Can you give me an advice?
My current plan is to add `.tralbum-about`, `.tralbum-credits` and `#license`.

**Review by FireMasterK — APPROVED** [review]

**Review by TobiGr — APPROVED** [review]

**Comment by fynngodau** [comment]
The copyright text is debatable because the individual tracks also always have copyright info, so it is not strictly necessary here and could be removed. The credits otoh should be read out here, as long as there's no separate method `getCredits`.

**Pull request merged**